Testing/ORKTest/ORKTest/MainViewController.m

```diff
@@ -2244,6 +2248,366 @@ - (void)optionalFormButtonTapped:(id)sender {
+    }
+    
+    {
+        // If we arrived at question_01_fail then fall through to question 2
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_02"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_01_fail"];
+    }
+    
+    {
+        // If we answer 'Yes' to Question 2, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_02"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForChoiceQuestionResultWithResultSelector:resultSelector expectedAnswerValue:@"Choose Yes"];
+    
+        ORKPredicateStepNavigationRule *predicateRule = [[ORKPredicateStepNavigationRule alloc] initWithResultPredicates:@[predicateQuestion]
+                                                                                          destinationStepIdentifiers:@[@"question_02_pass"]];
+        [task setNavigationRule:predicateRule forTriggerStepIdentifier:@"question_02"];
+    }
+    
+    {
+        // If we arrived at question_02_fail then fall through to question 3
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_03"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_02_fail"];
+    }
+    
+    
+    {
+        // If we answer 'Yes' to Question 3, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_03"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForChoiceQuestionResultWithResultSelector:resultSelector expectedAnswerValues:[NSArray arrayWithObjects: @"Cat",@"Dog", nil]];
+        
+        ORKPredicateStepNavigationRule *predicateRule = [[ORKPredicateStepNavigationRule alloc] initWithResultPredicates:@[predicateQuestion]
+                                                                                              destinationStepIdentifiers:@[@"question_03_pass"]];
+        [task setNavigationRule:predicateRule forTriggerStepIdentifier:@"question_03"];
+    }
+    
+    {
+        // If we arrived at question_03_fail then fall through to question 4
+        ORKDirectStepNavigationRule *directRule = nil;
+        directRule = [[ORKDirectStepNavigationRule alloc] initWithDestinationStepIdentifier:@"question_04"];
+        [task setNavigationRule:directRule forTriggerStepIdentifier:@"question_03_fail"];
+    }
+    
+    
+    {
+        // If we answer 'Yes' to Question 4, then proceed to the pass screen
+        ORKResultSelector *resultSelector = [ORKResultSelector selectorWithResultIdentifier:@"question_04"];
+        NSPredicate *predicateQuestion = [ORKResultPredicate predicateForChoiceQuestionResultWithResultSelector:resultSelector matchingPattern:@"Cat*"];
```

Comment:
These patterns use the same format accepted by `NSRegularExpression` (see [here](https://www.raywenderlich.com/30288/nsregularexpression-tutorial-and-cheat-sheet) for more info).

Hence, the proper expression is: `@"Cat.*"`.

`.` means *any character*, and `*` mans *the previous character repeated 0 or more times* .